test/integration/targets/tower_inventory_source_update/tasks/main.yml

```diff
@@ -0,0 +1,103 @@
+---
+- name: Run against a non-existant inventory
+  tower_inventory_source_update:
+    inventory: "DNE"
+    source: "DNE"
+  register: result
+  ignore_errors: True
+
+- assert:
+    that:
+      - "result is failed"
+      - "'Failed to find inventory' in result.msg"
+
+# This is directly lifted from tower_inventory_source tests
+- name: Add a Tower credential
+  tower_credential:
+    description: Credentials for Openstack Test project
+    name: openstack-test-credential
+    kind: openstack
+    organization: Default
+    project: Test
+    username: admin
+    host: https://example.org:5000
+    password: passw0rd
+    domain: test
+
+- name: Add a Tower inventory
+  tower_inventory:
+    description: Test inventory
+    name: openstack-test-inventory
+    organization: Default
+
+- name: Create a source inventory
+  tower_inventory_source:
+    name: source-test-inventory
+    description: Source for Test inventory
+    inventory: openstack-test-inventory
+    credential: openstack-test-credential
+    overwrite: True
+    update_on_launch: True
+    source_vars: "---\nprivate: false"
+    source: openstack
+  register: result
+
+- assert:
+    that:
+      - "result is successful"
+
+- name: Run againsta a non-existant inventory source
+  tower_inventory_source_update:
+    inventory: openstack-test-inventory
+    source: "DNE"
+  register: result
+  ignore_errors: True
+
+- assert:
+    that:
+      - "result is failed"
+      - "'Failed to find inventory source' in result.msg"
+
+- name: Run the inventory source update
+  tower_inventory_source_update:
+    inventory: openstack-test-inventory
+    source: source-test-inventory
+  register: result
+  ignore_errors: True
```

Comment:
If you wanted to test with a successful import, I think you could do this by creating a custom script, or you could do an SCM source. Since those are scripts, you can hard-code the inventory content and verify that it is successful.